render/id.go

```diff
@@ -66,7 +65,8 @@ func externalNodeID(n report.Node, addr string, local report.Networks) (string,
 
 	// If the dstNodeAddr is not in a network local to this report, we emit an
 	// internet pseudoNode
-	if ip := net.ParseIP(addr); ip != nil && !local.Contains(ip) {
+	var into [5]byte // one extra byte to save a memory allocation in critbitgo
+	if ip := report.ParseIPb([]byte(addr), into[:4]); ip != nil && !local.Contains(ip) {
```

Comment:
Why isn't the into allocation performed all the way inside report.parseIPv4? Do we end up with a heap instead of stack allocation when doing so? That should be explained in the comment.

Reply:
Yes, `into` is on the stack of `externalNodeID`; if it were declared in `ParseIP` it would escape to the heap.